python/ray/train/trainer.py

```diff
@@ -141,25 +140,6 @@ def _send_next_checkpoint_path_to_workers(self):
     def _run_with_error_handling(self, func: Callable):
         try:
             return func()
-        except TrainingWorkerError:
-            # Workers have already been restarted.
-            logger.info(
-                "Workers have been successfully restarted. Resuming "
-                "training from latest checkpoint."
-            )
```

Comment:
Then, this block never gets reached because we raised a different error type already in `_increment_failures`